sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -520,12 +522,14 @@ String name() {
     List<TopicPartition> partitions =
         Preconditions.checkArgumentNotNull(source.getSpec().getTopicPartitions());
 
-    // Each source has a single unique topic.
     for (TopicPartition topicPartition : partitions) {
-      this.kafkaTopic = topicPartition.topic();
-      break;
+      this.kafkaTopics.add(topicPartition.topic());
     }
 
+    LOG.info(
+        "Reader {} is reading from topics {}",
+        this.name,
+        kafkaTopics.stream().collect(Collectors.joining(", ", "{", "}")));
```

Comment:
The default `toString` implementation for a `Collection` would return `[e1, e2, e3]`, but this is fine if you want exactly the specified formatting. :)